feature/security/gnsi/credentialz/tests/ssh_public_key_authentication/ssh_public_key_authentication_test.go

```diff
@@ -0,0 +1,362 @@
+	}
+}
+
+func sendCredentialsRequest(t *testing.T, dut *ondatra.DUTDevice, request *credentialz.RotateAccountCredentialsRequest) {
+	credzClient := dut.RawAPIs().GNSI(t).Credentialz()
+
+	credzRotateClient, err := credzClient.RotateAccountCredentials(context.Background())
+	if err != nil {
+		t.Fatalf("failed fetching credentialz rotate account credentials client, error: %s", err)
+	}
+
+	err = credzRotateClient.Send(request)
+	if err != nil {
+		t.Fatalf("failed sending credentialz rotate account credentials request, error: %s", err)
+	}
+
+	_, err = credzRotateClient.Recv()
+	if err != nil {
+		t.Fatalf("failed receiving credentialz rotate account credentials request, error: %s", err)
+	}
+
+	err = credzRotateClient.Send(&credentialz.RotateAccountCredentialsRequest{
+		Request: &credentialz.RotateAccountCredentialsRequest_Finalize{
+			Finalize: request.GetFinalize(),
+		},
+	})
+	if err != nil {
+		t.Fatalf("failed sending credentialz rotate account credentials finalize request, error: %s", err)
+	}
+}
+
+func createNativeRole(t testing.TB, dut *ondatra.DUTDevice, role string) {
+	t.Helper()
+	switch dut.Vendor() {
+	case ondatra.NOKIA:
+		roleData, err := json.Marshal([]any{
+			map[string]any{
+				"services": []string{"cli", "gnmi"},
+			},
+		})
+		if err != nil {
+			t.Fatalf("Error with json Marshal: %v", err)
+		}
+
+		userData, err := json.Marshal([]any{
+			map[string]any{
+				"password": password,
+				"role":     []string{role},
+			},
+		})
+		if err != nil {
+			t.Fatalf("Error with json Marshal: %v", err)
+		}
+
+		SetRequest := &gpb.SetRequest{
+			Prefix: &gpb.Path{
+				Origin: "native",
+			},
+			Replace: []*gpb.Update{
+				{
+					Path: &gpb.Path{
+						Elem: []*gpb.PathElem{
+							{Name: "system"},
+							{Name: "aaa"},
+							{Name: "authorization"},
+							{Name: "role", Key: map[string]string{"rolename": role}},
+						},
+					},
+					Val: &gpb.TypedValue{
+						Value: &gpb.TypedValue_JsonIetfVal{
+							JsonIetfVal: roleData,
+						},
+					},
+				},
+				{
+					Path: &gpb.Path{
+						Elem: []*gpb.PathElem{
+							{Name: "system"},
+							{Name: "aaa"},
+							{Name: "authentication"},
+							{Name: "user", Key: map[string]string{"username": username}},
+						},
+					},
+					Val: &gpb.TypedValue{
+						Value: &gpb.TypedValue_JsonIetfVal{
+							JsonIetfVal: userData,
+						},
+					},
+				},
+			},
+		}
+		gnmiClient := dut.RawAPIs().GNMI(t)
+		if _, err = gnmiClient.Set(context.Background(), SetRequest); err != nil {
+			t.Fatalf("Unexpected error configuring User: %v", err)
+		}
+	default:
+		t.Fatalf("Unsupported vendor %s for deviation 'deviation_native_users'", dut.Vendor())
+	}
+}
+
+func setupUser(t *testing.T, dut *ondatra.DUTDevice) {
+	auth := &oc.System_Aaa_Authentication{}
+	user := auth.GetOrCreateUser(username)
+
+	if deviations.SetNativeUser(dut) {
+		// probably all vendors need to handle this since the user should have a role attached to
```

Comment:
Does configuring a role matter for testing authentication? For example, let's say my shell in /etc/passwd was /sbin/nologin - I couldn't get an interactive shell, but ssh -vv would still tell me if my authentication was successful.

Reply:
Removed vendor specific role settings